### ISSUE TITLE
Java modules

### ISSUE THREAD

**Issue by cavallium** [JanusGraph/janusgraph#3493]
**Describe the feature:**
Since Janusgraph supports java 11, it should work with java modules.
Currently, adding janusgraph.core and janusgraph.driver causes a "split package" error during compilation, because both jars have some package names in common.

**Describe a specific use case for the feature:**
Currently it's impossible to use janusgraph in any modularized java project, because split packages are strictly forbidden.

**Comment by matthewlowry** [comment]
v0.6.x still requires Java 8. On master Java 11 is supported and Java 8 support is dropped, but unless you are running off the master branch or a 1.x release candidate you will need to stay on Java 8.

**Comment by FlorianHockmann** [comment]
> and Java 8 support is dropped

Not quite, we only officially add support for Java 11 in version 1.0.0 that is why it will still support Java 8 and we currently plan to drop support for Java 8 only in a subsequent major release: #3547. Otherwise we would force people to move from Java 8 to 11 together with the update to JanusGraph 1.0.0.

**Comment by cavallium** [comment]
> Otherwise we would force people to move from Java 8 to 11 together with the update to JanusGraph 1.0.0.

Resolving split-packages in 1.0.0 would allow janusgraph to work as an automatic module in java 11+, without explicitly supporting modules, maintaining compatibility with java 8.
